frontend/src/app/forms/forms.page.html

```diff
@@ -20,38 +26,50 @@
 </ion-header>
 
 <app-side-menu
-  [sideMenuList]="sideMenuList"
-  (chosenStep)="navigateToStep($event)"
->
-</app-side-menu>
+  [formsData]="formsData"
+  [currentStep]="currentStep"
+></app-side-menu>
 
 <ion-content>
   <div class="main-container with-bottom-buttons">
     <div class="content">
-      <ng-container [ngSwitch]="currentMenu.selector">
+      <ng-container [ngSwitch]="currentStep">
         <form-basic-info
-          (updatedMenuCompletionStatus)="updateProgessStatus()"
-          [menuObject]="currentMenu"
-          [basicInfoObject]="overallInfo.controls.basicInfo"
-          *ngSwitchCase="'form-basic-info'"
+          [formsData]="formsData"
+          *ngSwitchCase="APPLICATION_STEP.BasicInformation"
         >
         </form-basic-info>
         <form-contact-info
-          (updatedMenuCompletionStatus)="updateProgessStatus()"
-          [menuObject]="currentMenu"
-          [contactInfoObject]="overallInfo.controls.contactInfo"
-          *ngSwitchCase="'form-contact-info'"
+          [formsData]="formsData"
```

Comment:
Why the whole formsData should be sent to component. Shouldn't be enough if only required step info is passed?

Reply:
Please have a look at contact-info.component.html. Here, it says "Nice to meet you <name>" (if a name has already been entered). I was able to easily achieve this since I had access to the whole formsData object. Another example: Please have a look at submit-page.component.html Depending on whether the form is valid or not, a different text is displayed.

So, by passing the whole formsData object to the individual components, you can easily create all kinds of "advanced" interactions utilizing form data entered in different steps.

I don't see any disadvantage from passing the whole object to the components (you just have to write "formsData.controls" pretty often :D ). But the advantage is that you can easily implement all kinds of interactions that wouldn't be possible if we had only one "slice" of this object.

I'd suggest to pass this object to all components and never to use the individual step info data. This allows us to easily add fancy stuff easily without having to refactor the whole component just because I want to display the user's name ;)